File: src/models/port-forwarding.ts
export interface PortForwarding {
	id: string;
	host_id: string;
	type: "local" | "remote" | "dynamic";
	local_port: number;
	remote_host?: string;
	remote_port?: number;
	description?: string;
	enabled: boolean;
	created_at: Date;
	updated_at: Date;
	is_pro_feature: boolean;
}

File: src-tauri/src/models/port_forwarding.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub enum PortForwardingType {
    #[serde(rename = "local")]
    Local,
    #[serde(rename = "remote")]
    Remote,
    #[serde(rename = "dynamic")]
    Dynamic,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct PortForwarding {
    pub id: String,
    pub host_id: String,
    #[serde(rename = "type")]
    pub forwarding_type: PortForwardingType,
    pub local_port: u16,
    pub remote_host: Option<String>,
    pub remote_port: Option<u16>,
    pub description: Option<String>,
    pub enabled: bool,
    pub created_at: i64,
    pub updated_at: i64,
    pub is_pro_feature: bool,
} 